src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extensions.cs

```diff
@@ -0,0 +1,539 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal partial class SourceNamedTypeSymbol
```

Comment:
Consider introducing a specialized derived type in order to avoid adding extension specific fields for all types.